pipelines/_ingredients/ApPipe.yaml

```diff
@@ -135,3 +135,6 @@ contracts:
   - contract: transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceTable.name ==
               diaPipe.connections.ConnectionsClass(config=diaPipe).diaSourceTable.name
     msg: "transformDiaSrcCat.diaSourceTable != diaPipe.diaSourceTable"
+  - contract: detectAndMeasure.connections.ConnectionsClass(config=detectAndMeasure).spatiallySampledMetrics.name ==
+              diffimTaskPlots.connections.ConnectionsClass(config=diffimTaskPlots).data.name
+    msg: "detectAndMeasure.spatiallySampledMetrics != diffimTaskPlots.spatiallySampledMetrics"
```

Comment:
So that people don't get lost when making future edits, please keep the contracts sorted as described in the comment. In this case, it would go with the other `detectAndMeasure` contracts, probably at the end (though `diffimTaskPlots` subverts the assumption of a linear pipeline...).

Reply:
Can you also add contracts for `analyzeAssocDiaSrcCore`, `analyzeTrailedDiaSrcCore`, and `diffimTaskCore`? The same bug could potentially happen there.

Reply:
Yes, though `diffimTaskCore` is weird because it reads the task metadata instead of the output data products. That does not get written with any of our parameters like `fakesType` or `coaddName`.

Reply:
Good point, a contract on the metadata would be redundant with the task labels already imposed by the pipeline.